### PR TITLE
Ensure all fi_ops_* structures are fully filled out to avoid seg faults.

### PR DESCRIPTION
**Pull request by bturrubiates**
The eq_test.c was seg faulting with the psm provider because the fabric ops structure had an uninitialized eq_open field. I just went through and initialized all fields to prevent that from happening somewhere else.

Also added an fi_no_scalable_ep implementation, and updated the comments in fi_enosys.h.

Signed-off-by: Ben Turrubiates <bturrubiates@lanl.gov>

### CODE REVIEW

File: include/fi_enosys.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ int fi_no_ops_open(struct fid *fid, const char *name,
 static struct fi_ops_fabric X = {
 	.size = sizeof(struct fi_ops_fabric),
 	.domain = fi_no_domain,
-	.endpoint = fi_no_passive_ep,
+	.passive_ep = fi_no_passive_ep,
```
Comment (shefty) [inline include/fi_enosys.h]: How in the world did this ever compile?

File: include/fi_enosys.h [hunk 2]
```diff
@@ -150,7 +150,7 @@ int fi_no_atomic_compwritevalid(struct fid_ep *ep,
 /*
 static struct fi_ops_cm X = {
 	.size = sizeof(struct fi_ops_cm),
-	.getname = X,
+	.getname = fi_no_getname,
```
Comment (shefty) [inline include/fi_enosys.h]: I don't think the intent of what I had in mind came across clearly.  The intent is that these comments can be copied directly into a provider's source code.  The places where the functions are set to 'X' must be defined by the provider.  These functions are not optional.

File: include/fi_enosys.h [hunk 3]
```diff
@@ -170,23 +170,12 @@ int fi_no_reject(struct fid_pep *pep, fi_connreq_t connreq,
...
-static struct fi_ops_av X = {
-	.size = sizeof(struct fi_ops_av),
-	.insert = X,
-	.insertsvc = X,
-	.insertsym = X,
-	.remove = X,
-	.lookup = X,
-	.straddr = X,
-};
-*/
```
Comment (shefty) [inline include/fi_enosys.h]: same comment as above

File: include/fi_enosys.h (hunk 2 above)
Comment (bturrubiates) [inline include/fi_enosys.h]: My bad. The fi_no_getname function already existed so I assumed that the comment hadn't been updated. I retract my statement about one of the providers not implementing .getname, I was mistaken.

File: include/fi_enosys.h (hunk 3 above)
Comment (bturrubiates) [inline include/fi_enosys.h]: I just repositioned the fi_ops_av and filled it in with the fi_no_* that were available. If those weren't supposed to be there then I can update the pull request. I'm not exactly sure what is required/not required so I just went with what was already defined.

**Comment by shefty** [comment]
@j-xiong @goodell @jithinjosepkl - can you please review the provider changes before I merge?

**Comment by j-xiong** [comment]
The PSM changes look fine to me.

A side question is: what is the “ops_open” field of fi_ops supposed to do? I don’t catch the moment when  it was added.

**Comment by jithinjosepkl** [comment]
Sockets look fine, But, I will remove the fi_no_tagged_injectdata to actual implementation in a following PR.

**Comment by shefty** [comment]
ops_open is used to export provider specific operations.

**Comment by jsquyres** [comment]
There seems to be a problem with the declaration of fi_no_cq_readfrom -- does it have the wrong prototype?

```
prov/usnic/src/usdf_cq.c:656:15: warning: incompatible pointer types
      initializing 'ssize_t (*)(struct fid_cq *, void *, size_t, fi_addr_t *,
      const void *, int)' with an expression of type 'ssize_t (struct fid_cq *,
      void *, size_t, fi_addr_t *)' [-Wincompatible-pointer-types]
        .sreadfrom = fi_no_cq_readfrom,
                     ^~~~~~~~~~~~~~~~~
```

**Comment by jsquyres** [comment]
No, I'm a bozo -- I see the error: it was just a typo.  Additional @bturrubiates PR coming shortly...

**Comment by jsquyres** [comment]
Or, more specifically, I added the fix to bturrubiates/libfabric#4

**Comment by bturrubiates** [comment]
Good catch @jsquyres. Thanks for building that...I'm only able to build sockets, verbs, and psm.

**Comment by jsquyres** [comment]
[hub](https://hub.github.com/) is my friend.  :-)

**Comment by jsquyres** [comment]
:+1:

**Comment by bturrubiates** [comment]
Neat. I'll look into that. Thanks for the link @jsquyres.

**Pull request merged**